hw_numpy/numpy_challenge.py

```diff
@@ -0,0 +1,43 @@
+if __name__ == "__main__":
+  import numpy as np
+  a = np.ones((3, 5))
+  b = np.arange(0, 60, 5).reshape(3, 4)
+  c = np.random.random((10, 10))
+
+# matrix multiplication
+def matrix_multiplication(first_arr, second_arr):
+   if first_arr.shape[1] == second_arr.shape[0]:
+     return(np.dot(first_arr, second_arr))
+
+# check if it is possible to multiply arrays
+def multiplication_check(arr):
+  b = np.vectorize(lambda x: x.shape)(np.array(list(map(lambda x: np.array(x), arr))))
+  if all(b[0][1:] == b[1][:-1]) == True:
+    return True
+  else:
+    return False
```

Comment:
1. Плохой нейминг переменных, совершенно непонятно, что такое `b`. А по имени `arr` вообще можно подумать, что это один эррей, а не их список

Reply:
2. \+ лишние условия и логические операции
3. Немного плохо читается, я бы разбил на несколько строк
```suggestion
def multiplication_check(matrices_list):
  combined_array = np.array(list(map(lambda x: np.array(x), matrices_list)))
  shapes_array = np.vectorize(lambda x: x.shape)(combined_array)
  return all(shapes_array[0][1:] == shapes_array[1][:-1])
```